cms/core/tests/test_views.py

```diff
@@ -32,3 +33,25 @@ def test_google_tag_manager_script_present(self):
 
         self.assertIn("https://www.googletagmanager.com/gtm.js?id=", response.rendered_content)
         self.assertIn(settings.GOOGLE_TAG_MANAGER_CONTAINER_ID, response.rendered_content)
+
+
+class ReadinessProbeTestCase(SimpleTestCase):
+    """Tests for the readiness probe endpoint."""
+
+    url = reverse("internal:readiness")
+
+    def test_get(self):
+        """Check the ready endpoint works with a GET request."""
+        response = self.client.get(self.url)
+
+        self.assertEqual(response.status_code, 204)
+        self.assertEqual(response.content, b"")
+        self.assertEqual(response.templates, [])
+
+    def test_head(self):
+        """Check the ready endpoint works with a HEAD request."""
+        response = self.client.head(self.url)
+
+        self.assertEqual(response.status_code, 204)
+        self.assertEqual(response.content, b"")
+        self.assertEqual(response.templates, [])
```

Comment:
question: seeing that the tests look at GET/HEAD, should the view forbid all the other methods?

Reply:
I explicitly want `GET` and `HEAD`, since that's what the probes and/or a test `curl` will use. I don't really care about other methods.

We could restrict the method, but I thought it's not worth the extra complexity (sure, there's a decorator), or miniscule performance overhead (premature optimisation, I know).

Reply:
From a security perspective, we should avoid leaving flows untested. We should restrict to GET/HEAD if that's all we care about.